Artifacts/MsSql/Structure/Security/0040-Tables - RLPhase2.sql

```diff
@@ -0,0 +1,132 @@
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ClaimSets_ClaimSet_ClaimSetId]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ResourceClaims_ResourceClaim_ResourceClaimId] FOREIGN KEY([ResourceClaim_ResourceClaimId])
+REFERENCES [dbo].[ResourceClaims] ([ResourceClaimId])
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizations_dbo.ResourceClaims_ResourceClaim_ResourceClaimId]
+GO
+
+---------------------------- [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] --------------------------------------------------------
+
+
+CREATE TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides](
+	[ClaimSetResourceClaimActionAuthorizationStrategyOverrideId] [int] IDENTITY(1,1) NOT NULL,
+	[AuthorizationStrategy_AuthorizationStrategyId] [int] NOT NULL,
+	[ClaimSetResourceClaimActionAuthorization_ClaimSetResourceClaimActionAuthorizationId] [int] NOT NULL,
+ CONSTRAINT [PK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides] PRIMARY KEY CLUSTERED 
+(
+	[ClaimSetResourceClaimActionAuthorizationStrategyOverrideId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.AuthorizationStrategies_AuthorizationStrategy_Authorization] FOREIGN KEY([AuthorizationStrategy_AuthorizationStrategyId])
+REFERENCES [dbo].[AuthorizationStrategies] ([AuthorizationStrategyId])
+ON DELETE CASCADE
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.AuthorizationStrategies_AuthorizationStrategy_Authorization]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.ClaimSetResourceClaimActionAuthorizations_ClaimSetResourceC] FOREIGN KEY([ClaimSetResourceClaimActionAuthorization_ClaimSetResourceClaimActionAuthorizationId])
+REFERENCES [dbo].[ClaimSetResourceClaimActionAuthorizations] ([ClaimSetResourceClaimActionAuthorizationId])
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.ClaimSetResourceClaimActionAuthorizations_ClaimSetResourceC]
+GO
+
+
+------------------------------ [dbo].[ResourceClaimActionAuthorizations] -------------------------------------------------
+CREATE TABLE [dbo].[ResourceClaimActionAuthorizations](
```

Comment:
Here, I think we should add a unique constraint on `ResourceClaim_ResourceClaimId` and `Action_ActionId`. We should not be able to define metadata for an action more than once for each resource claim.

@vimayya Thoughts?